src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/ToolRestoreUtility.cs

```diff
@@ -18,9 +19,10 @@ public static class ToolRestoreUtility
         /// its own project. For now, we always restore for a null runtime and a single
         /// constant framework.
         /// </summary>
-        public static PackageSpec GetSpec(string projectFilePath, string id, VersionRange versionRange, NuGetFramework framework)
+        public static PackageSpec GetSpec(string projectFilePath, string id, VersionRange versionRange, NuGetFramework framework, string packagesPath, List<string> fallbackFolders, List<PackageSource> sources, WarningProperties projectWideWarningProperties)
+
         {
-            var name = $"{id}-{Guid.NewGuid().ToString()}";
+            var name = $"{id}-{framework.Framework}-{versionRange.ToNormalizedString()}";
```

Comment:
write a helper method to handle creating this, I would also lowercase it if this is part of the hash to avoid casing issues between projects